core.js

```diff
@@ -2894,7 +2916,48 @@ class HTMLScriptElement extends HTMLLoadableElement {
         this.ownerDocument.defaultView.fetch(url)
           .then(res => {
             if (res.status >= 200 && res.status < 300) {
-              return res.text();
+              // when script type is set to 'application/require', fetch refuses to fetch.
+              //if (this.getAttribute('type') == 'application/require')
+              if (_shouldRequire(url)) {
+                if (process.env.DEBUG) console.log('require: ' + url);
+                return new Promise((accept, reject) => {
+                  res.arrayBuffer().then(ab => {
+                    const data = new Uint8ClampedArray(ab);
+                    const filename = path.basename(url);
+                    const name = filename.split('.')[0];
+                    const dst = path.join(SITE_MODULES, filename);
+                    fs.writeFileSync(dst, data);
+                    const decompress = require('decompress');
+                    var writer = fs.createWriteStream(dst);
+                    writer.write(Buffer.from(ab));
+                    writer.on('finish', () => {
+                      decompress(dst, SITE_MODULES).then(files => {
+                        const dir = path.join(SITE_MODULES, name);
+                        try {
+                          rmrf(dir);
```

Comment:
This is quite bad for performance. We shouldn't do anything that blocks the main thread, especially on loading. This will basically hang VR depending on the disk speed, beyond what is actually needed to load the file.

Reply:
I think there's some confusion here. `decompess` doesn't block the main thread -- the only thing that does is the `rmrf`, and 99% of the time, `dir` does not exist. The rmrf is there to ensure that it's possible to retry a failed require, which is a very rare case.